pkg/controller/atlasdeployment/advanced_deployment.go

```diff
@@ -210,12 +212,16 @@ func handleAutoscaling(ctx *workflow.Context, desiredDeployment *mdbv1.AdvancedD
 
 // MergedAdvancedDeployment will return the result of merging AtlasDeploymentSpec with Atlas Advanced Deployment
 func MergedAdvancedDeployment(atlasDeploymentAsAtlas mongodbatlas.AdvancedCluster, specDeployment mdbv1.AdvancedDeploymentSpec) (mergedDeployment mdbv1.AdvancedDeploymentSpec, atlasDeployment mdbv1.AdvancedDeploymentSpec, err error) {
+	if IsFreeTierAdvancedDeployment(&atlasDeploymentAsAtlas) {
+		atlasDeploymentAsAtlas.DiskSizeGB = nil
```

Comment:
Just in case it is possible to have `512 MB` for this field in places other than free tier I suggest to actually check the value and make the decision based on that. We technically don't even need to check for the actual `InstanceSize` value.

```
if spec == nil && atlas < 1 { make nil }
```

Reply:
M0 supports only 512mb disk size

Reply:
Ok:)